featurizers/fingerprints.py

```diff
@@ -386,15 +391,15 @@ def _calculate_fingerprint(
                 },
             )
 
-            # Set a property for each fingerprint, to be able to obtain energy later
-            for i in range(len(fps)):
-                fps[i].set_prop("Energy", energies[i])
-
-                if self.is_folded:
-                    fps[i] = fps[i].fold(self.fold_bits)
+            energies = values[2]
+            fp = fps[np.argmin(energies)]
 
-            result.extend(fps)
+            if self.is_folded:
+                fp = fp.fold(self.fold_bits)
```

Comment:
I wonder if we should allow non-folded version at all. After all, we assume Scikit-learn interface, which does require Numpy or Scipy matrices.

Reply:
`fprints_from_mol()` takes number of bits as an argument and limits the upper bound of the output tensor to that number. Folding is then performed on vectors of already limited and uniform size